### DISCUSSION TITLE
Attribute Error While Running Trame ParaView

### DISCUSSION THREAD

**Discussion by ghost** [Kitware/trame#553] [General] [answered]
Hello, I'm trying to run a Trame app which references ParaView, but it shows AttributeError: 'Coroutine' object has no attribute 'on_message'. Any help will be appreciated

Here is more info of my venv:
- Anaconda venv
- Python 3.9
- ParaView 5.11.2
- Conda installed paraview (also 5.11.2), trame, trame-vtk, trame-vuetify

![image](https://github.com/Kitware/trame/assets/93155074/ea57da12-bc0a-49de-a423-b4fe0c7f97dd)

**Comment by jourdain** [comment]
Conda recipe have some cross version dependencies issue.
You probably need to update trame, wslink, trame-server, trame-client or add more constraints on the various versions.

**Reply by ghost** [reply]
I updated all above packages and get a new error about numpy RunTimeError: This module depends on numpy module. Please make sure it is installed properly.
![image](https://github.com/Kitware/trame/assets/93155074/936c4c4b-5d1b-4365-8725-26a0454d2a9f)

Here is my venv and dependencies:
![image](https://github.com/Kitware/trame/assets/93155074/2784de55-40fe-485c-8843-37f9d1c4c650)

**Reply by jourdain** [reply]
numpy<2

**Reply by ghost** [reply]
Thank you for your help. I installed numpy=1.26.2, but it still shows the same error.

**Reply by jourdain** [reply]
Not sure what could be wrong with your Python env.

**Reply by ghost** [reply]
Okay. Thank you for your help anyway :)

**Reply by ghost** [reply]
I used pip installed numpy=1.26.2 instead of conda install and it doesn't show numpy error anymore, but it goes back to the coroutine error :)

![image](https://github.com/Kitware/trame/assets/93155074/2f2ff9a6-bac3-4569-a61e-8d024cd79798)

**Reply by jourdain** [reply]
Try to be consistent between pip and conda.

**Reply by jourdain** [reply]
Oh, you missed the most important part... You are using `trame.tools.serve`... Are you sure your app is designed to work with it?

**Reply by ghost** [reply]
I'm running the exact same remote rendering app with Paraview as #531 which used tram.tools.serve

**Reply by jourdain** [reply, marked as answer]
The issue "maybe" is that ParaView bring wslink<2 and your trame env is having wslink>=2.
Also I still don't know which code you are running and it is properly designed with a class encapsulation.

**Reply by ghost** [reply]
I changed to wslink<2 and that fixed the error. Thank you for your help.

**Reply by jourdain** [reply]
Ok so for those coming here that means installing `trame<3.6`
